cpu/src/lib.rs

```diff
@@ -379,10 +379,20 @@ where
         let mem_addr = fp + ops.a();
 
         // Read from the advice tape into memory
-        let advice_byte = advice.get_advice().expect("No more advice");
-        state
-            .mem_mut()
-            .write(clk, mem_addr as u32, Word::from_u8(advice_byte), true);
+        let advice_byte = advice.get_advice();
```

Comment:
nit: could be shorter if we used `advice.get_advice().unwrap_or(u32::MAX)` rather than branching below